### PR TITLE
Move the dashboard's component library from antd to material-ui

### LINKED ISSUE #1408: Switch web component library from Ant to Material-UI
This will be a tracking ticket for the move. 

Switch the component pattern library we use in the frontend from Ant to [Material-UI](https://github.com/mui-org/material-ui).

All of our current components and layouts are based on ant. We'll need to find the equivalents of those in material, and resolve any feature discrepancies between the two. From my looking at it thus far, 

Components that need to be switched over, in rough order of effort:
- Table - Sorting seems to be there, but I think we'll need to implement our own filtering https://material-ui.com/api/table/#table
- Layout
  - Grid layout  (Row, Col) - move to https://material-ui.com/layout/grid/
  - Sider - https://material-ui.com/demos/drawers/#drawer
  - Nav bar - move to https://material-ui.com/demos/app-bar/#app-bar
- Menu - https://material-ui.com/demos/menus/
- Form components  - https://material-ui.com/demos/selection-controls/#checkboxes
  - Switch
  - Radio
- Modal - https://material-ui.com/utils/modal/
- Tooltip - https://material-ui.com/demos/tooltips/
- Spin (spinner) - https://material-ui.com/demos/progress/#progress
- Button - https://material-ui.com/api/button
- Icon - https://material-ui.com/style/icons/

### PR DESCRIPTION
**Pull request by rmars**
Switch the component library from [antd](https://ant.design/) to [material-ui](https://material-ui.com/).

This branch changes all uses of antd components to their closest equivalent in material. There is still a lot of polish that needs to go into the look of individual components, but since the major component rewrites are done, I think it's time to put tis up for review and get it in so that further work can be done in smaller branches. 

## Changes in this branch
- replace all uses of antd with material-ui components
- slight modifications of eslint rules
- restructuring of app components to be rendered under the Navigation component
- deleted most of our css (replaced with material's inline styles)
- pinned package versions in `package.json` (mostly removing `^`)

## Follow-up work
Here is a list of known UI polish in the current branch (I will ticket each one, as well as the results of code-review of this branch, to be done in follow up branches):
- Navigation/Sidebar #1781
  - I removed the list of resources per namespace and the namespace dropdown in the sidebar added in #1500 for simplicity. we should re-add it. I restructured the Navigation to be part of a larger component which renders the page as a subcomponent, which should make future work on #1669 easier
  - Side navigation doesn’t go all the way down on the Overview page
  - The icons for the side navigation were chosen without a lot of thought. Revisit this, as done in #1637. Resources that can help: https://material.io/tools/icons/?icon=home&style=baseline, https://materialdesignicons.com/. You can also use FontAwesome: (https://material-ui.com/style/icons/#font-icons)
- Theming and colours (#1782): for the charts and app-bar I did a very cursory colour selection. 
  - We should review the colour choices and pick a coherent set  (https://material.io/tools/color should help)
  - The theming in `index.js` should be moved into its own file as we expand it
- Tables (#1783) throughout the app need to be revisited and neatened
  - fix table widths, currently the tables are too wide, the columns are too narrow
  - add sorting to the table columns
  - add filtering to the tables
- Popover (#1784): popover on src dst column in the top tables goes away too fast for you to click anything on it, clicking on the links doesn’t work
- Octopus graph (#1785)
  - unmeshed resources and collapsed resources need to be full column width
  - octopus arms need arrow alignment to new card widths
- TapQueryForm: need form validation
- Tooltips: add arrows to tooltips
-  Tooling: upgrade to babel7 (#1786)

Fixes #1408

### CODE REVIEW

File: web/app/js/components/ExpandableTable.jsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                        numeric={c.isNumeric}>
+                        {c.render(d)}
+                      </TableCell>
+                    ))
+                  }
+                </TableRow>
+              </React.Fragment>
+
+            );
+          })}
```
Comment (siggy) [inline web/app/js/components/ExpandableTable.jsx]: nit: indent

File: web/app/js/components/util/Progress.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import LinearProgress from '@material-ui/core/LinearProgress';
+import { withStyles } from '@material-ui/core/styles';
+
+const colorLookup = {
+  good: {
+    colorPrimary: '#c8e6c9', // background bar color (lighter)
```
Comment (siggy) [inline web/app/js/components/util/Progress.jsx]: per https://material-ui.com/style/color/, `#c8e6c9` is documented as `green[100]`:

```js
import red from '@material-ui/core/colors/green';
...
colorPrimary: green[100], // background bar color (lighter)
```

is it advisable/appropriate to reference them this way?

File: web/app/js/components/Navigation.jsx [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+    );
+  }
+  render() {
+    const { classes, ChildComponent } = this.props;
+
+    return (
+      <div className={classes.root}>
+        <AppBar
+          position="absolute"
+          className={classNames(classes.appBar, this.state.drawerOpen && classes.appBarShift)}>
```
Comment (dadjeibaah) [inline web/app/js/components/Navigation.jsx]: More of a readability nit: I am not entirely sure how `classNames` interprets statements like `this.state.drawerOpen && classes.appBarShift` but I found it a little difficult to comprehend. I tried rewriting the statement like so:
```jsx
className={classNames(classes.appBar, {[`${classes.appBarShift}`]: this.state.drawerOpen})}>
```
And that seemed to work the same way.

File: web/app/js/components/Navigation.jsx [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+      </MenuItem>
+    );
+  }
+  render() {
+    const { classes, ChildComponent } = this.props;
+
+    return (
+      <div className={classes.root}>
+        <AppBar
+          position="absolute"
```
Comment (dadjeibaah) [inline web/app/js/components/Navigation.jsx]: Could this line be moved to the `appBar` style object?

Comment (rmars) [reply web/app/js/components/Navigation.jsx]: This is part of `AppBar`'s API:
https://material-ui.com/api/app-bar/#appbar

File: web/app/js/components/Navigation.jsx (hunk 3 above)
Comment (rmars) [reply web/app/js/components/Navigation.jsx]: Hahahah, what that line is actually doing is a combination of cool es6 tricks:

First, it's using the `&&` operand to return the string name of the class if `drawerOpen` is true.
(The js `&&` operand will return the latter of the provided values if the first expression is truthy). Example:
```
>> false && "foo"
>> false

>> true && "foo"
>> "foo"
```

Then, it's using the fancy object assignment to assign `appBarShift`'s className to the object if provided:
```
>> let a = "a_string";
>> let obj = { a }

>> obj
{a: "a_string"}
```

So this is equivalent to providing an object whose key is `classes.appBarShift` and value is `this.state.drawerOpen` as you've written. I'll rewrite it to be less trick-y!

Comment (dadjeibaah) [reply web/app/js/components/Navigation.jsx]: Oh my gosh! so much sorcery! That is pretty cool. it's a TIOLI so feel free to do whatever you think is best.

Comment (rmars) [reply web/app/js/components/Navigation.jsx]: Hahaha, yeah I think readable > cool, so I'll change them to more traditional args. 
It's a pattern I see popping up a lot in the material examples, so maybe we can change later.

File: web/app/js/components/BaseTable.jsx [hunk 5]
```diff
@@ -1,31 +1,78 @@
...
-// https://github.com/ant-design/ant-design/blob/master/components/table/Table.tsx#L348
-export default class BaseTable extends Table {
-  constructor(props) {
-    super(props);
-    Table.prototype.toggleSortOrder = this.toggleSortOrder;
-  }
+const styles = theme => ({
+  root: {
+    width: '100%',
+    marginTop: theme.spacing.unit * 3,
```
[the remark this answers is not in the document]
Comment (rmars) [reply web/app/js/components/BaseTable.jsx]: done!

**Pull request merged**